app/models/claim_auto_approval.rb

```diff
@@ -25,7 +25,7 @@ def auto_approve!
         claim.notes.create!(body: "This claim has been marked for a quality assurance review")
       end
 
-      ClaimMailer.approved(claim).deliver_later unless claim.awaiting_qa?
+      claim.policy.mailer.approved(claim).deliver_later unless claim.awaiting_qa?
```

Comment:
so will not call `deliver_later` on
```
self.class.provider_approved(claim).deliver_later
```
does it work as expected? I feel like we still need something in between the mailer and whatever calls the mailer to handle the events lifecycle. I was able to get away with not doing it in a recent change. maybe this just crosses the line? tbh i'm not sure what this thing would look like though

Reply:
The emails are sent to the provider and practitioner as expected.

I think what we really want to do is
```
claim.policy.claim_approved(claim)
```
(and `claim.policy.claim_rejected(claim)` etc)
and let the policy decide what emails to send / other actions to take when a claim is approved. Not sure if this should be part of this pr or in a separate pr.